### ISSUE TITLE
win64 zombie renderer process after crashing

### ISSUE THREAD

**Issue by shahrouzz** [electron/electron#20461]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  * v6.0.11
* **Operating System:**
  * Microsoft Windows 10 Pro 10.0.18362 Build 18362

### Expected Behavior
No processes should be left behind after the main process crashes.

### Actual Behavior
A zombie renderer process is left behind after the crash.

### To Reproduce

run from cmd.exe:
```sh
$ git clone https://github.com/shahrouzz/electron-win64-zombie-bug.git
$ cd electron-win64-zombie-bug
$ yarn
$ yarn electron .
```
After the crash there will still be a electron.exe renderer process running in taskmanager.

note: if you run it in bash on windows, use the following command to run:

```sh
./node_modules/electron/dist/electron.exe .
```

### Additional Information
This is how we are crashing the process:
```js
setTimeout(function() {
  console.log("crashing!");
  remote.process.crash();
}, 1000);
```
Note that this only seem to happen if the crash triggered through the remote api. If process.crash() is called from the main process it works as expected.

It seems like processes that are spawned by BrowserViews also become zombies under certain circumstances, so the problem isn't unique to BrowserWindow. The general problem for us is that if the our native code and brings down the main process there are zombie processes left behind by electron.

**Comment by Slapbox** [comment]
As far as last working version for this issue, I've been seeing this issue since at least `1.7`. It's a bit of a pain, but I just close them via task manager when it happens. It can be pretty bad though when debugging some problems.

**Comment by yohan1234** [comment]
Could it be that the Renderer processes do not set the Main process as their parent? If they did, I believe that they should automatically terminate when the main process crashes. This would be something internal to Electron though and cannot be set via the API.

**Comment by miniak** [comment]
@yohan1234 Windows does not have the UNIX concept of parent / child process. This issue seems to only apply to non-sandboxed renderer. Sandboxed renderers are enclosed in a job object, which has `JOB_OBJECT_LIMIT_KILL_ON_JOB_CLOSE` set that makes sure the child process is killed when the parent is gone, which is the only process with an open handle to the job object.

**Comment by miniak** [comment]
I was trying fix it in #18153, but the Chromium change that this relied on got reverted as it broke CI builds (it was also failing Electron builds)

**Comment by shahrouzz** [comment]
I can't reproduce this issue in electron 6.1.7 and electron 8.0.0 so I assume it is fixed.

**Comment by Slapbox** [comment]
Also confirming this appears fixed since at least `6.1.x`.
